New EVA voice after deploying a building

FS-21 opened the pull request:
- You can now replace the current EVA voice when a specific building is placed/deployed.
- If any buiding is undeployed/sold/destroyed EVA voice will be evaluated again looking all the `NewEVAVoice.Index`.
- `NewEVAVoice.Index` is the index of the new EVA voice. Ares is hightly recomended because these indexes are reading the new section `[EVATypes]` at `evamd.ini` introduced by Ares. Look at Ares documentation regarding new EVA voices for more information.
- In case of multiple buildings with different `NewEVAVoice.Index` EVA voices then `NewEVAVoice.Priority` establish a priority queue, being the highest value the selected one.
- `NewEVAVoice.RecheckOnDeath` re-checks a new EVA voice after the destruction/undeployment of of of these buildings.
- `NewEVAVoice.InitialMessage` plays an EVA message to the player when a different EVA has been selected.

In `rulesmd.ini`:
```ini
[SOMEBUILDING]                    ; BuildingType
NewEVAVoice.Index=                ; integer
NewEVAVoice.Priority=1            ; integer
NewEVAVoice.RecheckOnDeath=false  ; boolean
NewEVAVoice.InitialMessage=       ; EVA entry
```

FS-21 committed:
Initial commit

FS-21 committed:
YRpp update

added 4 new GameStrings

FS-21 committed:
deleted a white space

FS-21 committed:
Merge branch 'develop' into feature/StructuresCanChangeEvaVoice

FS-21 committed:
Fixes

- The player will hear only his own EVA voices from the tag NewEVAVoice.InitialMessage.
- The readed data of the [EVATypes] was deleted before it was used by the new code.
- Small fixes & tweaks.

FS-21 committed:
Merge remote-tracking branch 'origin/develop' into feature/StructuresCanChangeEvaVoice

FS-21 committed:
YRpp

FS-21 committed:
Merge branch 'develop' into feature/StructuresCanChangeEvaVoice

FS-21 committed:
Merge branch 'develop' into feature/StructuresCanChangeEvaVoice

FS-21 committed:
Merge branch 'develop' into feature/StructuresCanChangeEvaVoice

Starkku commented on docs/User-Interface.md:
````diff
@@ -27,6 +27,24 @@ IngameScore.WinTheme=  ; Soundtrack theme ID
+In `rulesmd.ini`:
+```ini
+[SOMEBUILDING]                    ; BuildingType
+NewEVAVoice=                      ; boolean
+NewEVAVoice.Index=0               ; integer
````
Get rid of the bool and use `Nullable` or negative default value index.

Starkku commented on src/Ext/Rules/Body.cpp:
```diff
@@ -313,6 +316,10 @@ void RulesExt::ExtData::LoadBeforeTypeData(RulesClass* pThis, CCINIClass* pINI)
 	}
 }
 
+namespace EVAIndexListTemp
+{
+	std::vector<std::string> EVAIndexList;
+}
```
Is this namespace actually used anywhere?

FS-21 replied:
From initial tests, thanks for finding that garbage!

Starkku reviewed (changes requested):
The EVA voice change logic should be generalized and put in a separate function from buildings, probably in `SideExt` or something. Firestorm has a map trigger for this purpose and I suppose someone might want to reimplement that here as well.

Some smaller things to take note of as well.

FS-21 commented:
> The EVA voice change logic should be generalized and put in a separate function from buildings, probably in `SideExt` or something. Firestorm has a map trigger for this purpose and I suppose someone might want to reimplement that here as well.
> 
> Some smaller things to take note of as well.

Since Ares already has Map trigger actions for changing  these voices like in Firestorm I only focused in learning how to change it and apply this logic for buildings like in the video, not reimplementing again the EVA voices.

Let me see the SideExt you mentioned because I remember one of the first attempts was storing data in that place but things happened that made me move the code and I ended here.

FS-21 committed:
Applied feedback

- Removed NewEvaVoice tag.
- NewEvaVoice.Index now is nullable int and the checks are against this value instead of the old NewEvaVoice.

FS-21 committed:
Updated docs

FS-21 committed:
Merge remote-tracking branch 'origin/develop' into feature/StructuresCanChangeEvaVoice

FS-21 committed:
Applied feedback

Function UpdateMainEvaVoice(...) moved from Buildings to side class.

FS-21 committed:
Merge remote-tracking branch 'origin/develop' into feature/StructuresCanChangeEvaVoice

FS-21 committed:
Merge branch 'develop' into feature/StructuresCanChangeEvaVoice

FS-21 committed:
Merge branch 'develop' into feature/StructuresCanChangeEvaVoice

FS-21 committed:
Merge branch 'develop' into feature/StructuresCanChangeEvaVoice

FS-21 commented:
> Limbo Delivery buildings can't change EVA

Yes, I never tried that so I added support now.

I updated this code and added support when a building is captured by engineers.
